docs/src/views/Architecture.tsx

```diff
@@ -0,0 +1,224 @@
+        component state.
+      </li>
+      <li>
+        <strong>Accessibility keyboard handlers</strong> - provides the logic to handle components
+        navigation and focus.
+      </li>
+      <li>
+        <strong>Themes</strong> - the logic to inject and handle component visual representation
+        (theme).
+      </li>
+      <li>
+        <strong>Renderer</strong> - to render the HTML Output. Current implementation uses React,
+        but it can be extended to other frameworks.
+      </li>
+    </ul>
+    <p>
+      <img src="assets/stardust_arch_component_logic.png" />
+    </p>
+
+    <Header as="h3">Accessibility</Header>
+    <p>
+      One of the main purposes of the library is to make component accessible and provide ease
+      accessibility integration for the entire projects which will be consumer of the library.
+    </p>
+    <p>
+      The library encapsulates accessibility into <strong>Behaviors</strong>. Behavior is the
+      object, which has three major responsibilities:
+    </p>
+    <ul>
+      <li>
+        Define <code>aria-*</code> attributes for the standard components (e.g. Image, DropDown,
+        Dialog, etc), based on their state (e.g. active, disabled, expanded, etc).
+      </li>
+      <li>
+        Define the correct element <code>role</code>, if the HTML element tag differ from standard
+        (e.g. use <code>&lt;span&rt;</code> instead of <code>&lt;button&rt;</code>) to represent the
+        button element.
+      </li>
+      <li>
+        Define the keyboard navigation and focus handling for the composed components (e.g. List,
+        Grid, Radio Group, etc).
+      </li>
+    </ul>
+    <p>
+      Library provides the set of accessibility behaviors and keyboard handlers for the components
+      and allows to create and inject your own, if it is necessary for some custom components. In
+      most cases it should be enough to use the behaviors provided in the library, but, of course,
+      you can provide your own or extend existing behaviors.
+    </p>
+    <p>
+      In addition to behaviors and keyboard handlers, the toolkit also provides FocusZone component,
+      which can be used to add focus handling for the wrapped component. There is also FocusTrap
+      component, which allows to trap the focus inside some area, which is useful for such a
+      component as Dialog.
+    </p>
+
+    <Header as="h3">Theming</Header>
+    <p>
+      You can inject different themes into Stardust components (e.g. Light, Dark or High Contrast).
+      To achieve this, we use CSS in JS technique, which is provided by{' '}
+      <a href="http://fela.js.org/" target="_blank">
+        Fela library
+      </a>.
+    </p>
+    <p>
+      Each component has its own set of themes defined. Component logic handles theme switching and
+      style overriding by providing ability to set <strong>variables</strong> or override component
+      (or its children) styles using <code>styles</code> property.
```

Comment:
Components do not define themes, only the Provider defines a theme.  Component logic does not handle theme switching, all theme switching is handled solely by the user and what they pass to a theme.  All theming concerns are intentionally kept outside of the component.  Components are more like templates whose template slots are factories for creating components.

Components can override the theme by passing variables and styles as noted, however, this is duplicating what is already laid out in `/theming`.